src/app/shared/issue-pr-card/issue-pr-card-header/issue-pr-card-header.component.ts

```diff
@@ -18,14 +18,18 @@ export class IssuePrCardHeaderComponent {
   getOcticon() {
     const type = this.issue.issueOrPr;
     const state = this.issue.state;
+    const stateReason = this.issue.stateReason;
 
     switch (true) {
```

Comment:
Might be better to convert this into a regular if-else statement at this point with the increasing number of checks.

Reply:
I've made the changes! Though, would it be better to abstract out the checking logic into another service class instead because of the layers of nesting?